exporter/prometheusremotewriteexporter/wal.go

```diff
@@ -0,0 +1,247 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package prometheusremotewriteexporter
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"path/filepath"
+	"time"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/prometheus/prometheus/prompb"
+	"github.com/tidwall/wal"
+
+	"go.opentelemetry.io/collector/consumer/consumererror"
+)
+
+type walConfig struct {
+	Dir             string        `mapstructure:"path"`
+	CacheSize       int           `mapstructure:"cache-size"`
+	RefreshDuration time.Duration `mapstructure:"truncate-frequency"`
```

Comment:
For uniformness use underscore for word separation in config keys. That's what everything else in the Collector does.

Reply:
So I was asked to make it as uniform with Prometheus as possible, but sure, underscores it is then.